sources/xfscli/src/bin/fsck_xfs.rs

```diff
@@ -0,0 +1,198 @@
+use snafu::{OptionExt, ResultExt};
+use std::os::unix::fs::FileTypeExt;
+use std::path::Path;
+use std::process::exit;
+use std::process::Command;
+use std::{fs, str};
+
+use argh::FromArgs;
+
+use xfscli::error::{self, Error, Result};
+use xfscli::{perform_xfs_repair, XfsRepairResponseCode, BLKID, MOUNT, UMOUNT};
+
+// This CLI is rewritten in Rust for fsck.xfs shell script
+// https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/tree/fsck/xfs_fsck.sh
+// since there isn't a shell on Bottlerocket hosts.
+
+const REPAIR_MOUNT_DIR: &str = "/tmp/repair_mnt";
+struct KernelParams {
+    root: String,
+    root_flags: String,
+}
+
+/// check and repair a XFS file system
+#[derive(FromArgs)]
+struct Args {
+    /// auto repair the file system
+    #[argh(switch, short = 'a')]
+    auto_repair: bool,
+
+    /// auto check all file systems
+    #[argh(switch, short = 'A')]
+    auto_check: bool,
+
+    /// check the root filesystem as well when -A flag is set
+    #[argh(switch, short = 'p')]
+    check_root_fs: bool,
+
+    /// force repair filesystem
+    #[argh(switch, short = 'f')]
+    force: bool,
+
+    /// repair filesystem automatically
+    #[argh(switch, short = 'y')]
+    repair: bool,
+
+    /// mount-point | block-device
+    #[argh(positional)]
+    target: Option<String>,
+}
+
+fn main() {
+    let args: Args = argh::from_env();
+    match run(args) {
+        Ok(exit_code) => exit(exit_code),
+        Err(e) => {
+            eprintln!("{}", e);
+            exit(e.exit_code());
+        }
+    }
+}
+
+fn run(args: Args) -> Result<i32> {
+    let auto = args.auto_check || args.auto_repair || args.check_root_fs;
+    let force = args.force;
+    let repair = args.repair;
+
+    // This is diverted behavior from xfs shell script.
+    // In fsck no filesystems are specified on the command line,
+    // and the -A option is not specified,
+    // fsck will default to checking filesystems in /etc/fstab serially
+    // Hence device name is not a mandatory argument. But as there isn't a /etc/fstab
+    // file in Bottlerocket we have to specify the device.
+    let target = &args.target.as_ref().context(error::ParseTargetSnafu)?;
+
+    if !Path::new(target).exists() {
+        return Err(Error::DeviceNotFound {
+            target: target.to_string(),
+        });
+    }
+
+    if force {
+        // Preform xfs_repair
+        println!("Performing XFS repair..");
+        let mut repair_exit_code = perform_xfs_repair(vec!["-e", target]);
+
+        // Clear the log, mount and unmount the XFS file system
+        // before xfs_repair
+        if repair_exit_code == XfsRepairResponseCode::DirtyLogs && repair {
+            println!("Replaying log for {}", target);
+
+            // Using random directory (/tmp/repair_mntxxxx) instead of a targeted directory(/tmp/repair_mnt)
```

Comment:
nit, add a space here
```suggestion
            // Using random directory (/tmp/repair_mntxxxx) instead of a targeted directory (/tmp/repair_mnt)
```